### PR TITLE
Allow modifications of the request authorization headers

### PR DESCRIPTION
**Pull request by simonxmh**
This will let Avalanche make requests with custom authorization headers.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,7 +3,9 @@ WORKDIR $GOPATH/avalanche
 COPY . .
 RUN GO111MODULE=on CGO_ENABLED=0 GOOS=linux go build -o=/bin/avalanche ./cmd
 
-FROM scratch
+FROM alpine:latest
```
Comment (csmarchbanks) [inline Dockerfile:6]: What are you gaining from changing the dockerfile? It seems a bit orthogonal from the rest of the change.

File: cmd/avalanche.go [hunk 2]
```diff
@@ -31,6 +31,7 @@ var (
 	remoteRequestCount  = kingpin.Flag("remote-requests-count", "how many requests to send in total to the remote_write API.").Default("100").Int()
 	remoteReqsInterval  = kingpin.Flag("remote-write-interval", "delay between each remote write request.").Default("100ms").Duration()
 	remoteTenant        = kingpin.Flag("remote-tenant", "Tenant ID to include in remote_write send").Default("0").String()
+	remoteAuthHeader    = kingpin.Flag("remote-url-auth", "Authorization header to pass in for the remote-url").String()
```
Comment (csmarchbanks) [inline cmd/avalanche.go:34]: I would say just `remote-auth` for the flag, `url` does not add much. The help text should probably also specify that this is the value of the Authorization header.

**Review by csmarchbanks — COMMENTED** [review]
Thanks! A couple small comments/questions but otherwise LGTM.

File: Dockerfile (hunk 1 above)
Comment (simonxmh) [reply Dockerfile:6]: I need to add the ca-certificate module to establish tls communication with my remote targets.

**Review by shivasks — APPROVED** [review]

**Comment by igaskin** [comment]
Looks like this PR was already approved?  Is this feature still of interest? If not, may close as stale.
